__tests__/users.ts

```diff
@@ -330,6 +354,74 @@ describe('query userReadingRank', () => {
     });
   });
 
+  it('should return the reading rank with tags on version 2', async () => {
+    loggedUser = '1';
+    await con
+      .getRepository(Keyword)
+      .save([...keywordsFixture, ...additionalKeywords]);
+    await con.getRepository(PostKeyword).save(postKeywordFixtures);
+    await con.getRepository(View).save([
+      { userId: loggedUser, postId: 'p1', timestamp: lastWeekStart },
+      {
+        userId: loggedUser,
+        postId: 'p2',
+        timestamp: addDays(lastWeekStart, 1),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p3',
+        timestamp: addDays(lastWeekStart, 3),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p1',
+        timestamp: addDays(thisWeekStart, 1),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p2',
+        timestamp: addDays(thisWeekStart, 2),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p3',
+        timestamp: addDays(thisWeekStart, 3),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p4',
+        timestamp: addDays(thisWeekStart, 1),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p5',
+        timestamp: addDays(thisWeekStart, 2),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p6',
+        timestamp: addDays(thisWeekStart, 3),
+      },
+      {
+        userId: loggedUser,
+        postId: 'p7',
+        timestamp: addDays(thisWeekStart, 4),
+      },
+    ]);
+    const res = await client.query(QUERY, {
+      variables: { id: '1', version: 2 },
+    });
+    expect(res.errors).toBeFalsy();
+    expect(res.data.userReadingRank).toMatchSnapshot({
+      readToday: expect.anything(),
+      lastReadTime: expect.anything(),
+    });
+    const { tags } = res.data.userReadingRank;
+    expect(tags.length).toEqual(8);
+    const sum = tags.reduce((total, { readingDays }) => total + readingDays, 0);
+    expect(sum).toEqual(12);
```

Comment:
Why this is needed if we match a snapshot?

Reply:
Ah, yes. The snapshot overkills it. Let me keep the computation to make it easier to understand than looking at the snapshots.